protobuf/line_protocol.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+/*========================================================================*/
+/*                          ENUM DEFINTIONS                               */
+/*========================================================================*/
+// Definition of triggering mode: 
+enum Activation {
+  POLLING = 0;  //=> activation only on request (default)
+  EVENT = 1;    //=> activation only on events 
+  COMPLEX = 2;  //=> combination of Polling and Event-driven mode
+}
+// Definition of digital pin modes
+enum Digital_Mode{
+  INPUT = 0; // (default)
+  OUTPUT = 1;
+  INPUT_PULLUP = 2;
+}
+// Definition of digital pin output
+enum Digital_Output{
+  LOW = 0; // (default)
+  HIGH = 1;
+}
+
+// Definition of possible UART-TTL ports
+enum UART_Port{
+  UART2 = 0; // (default)
+  UART3 = 1;
+}
+
+/*========================================================================*/
+/*                  MAIN MESSAGES: REQUEST + FEEDBACK                     */
+/*========================================================================*/
+// message sent from controller to gateway
+message Feedback {
+  //int32 code = 1;
+  uint32 profile_id = 1;  //  used to identify profile
+  string message = 2;
```

Comment:
What will be stored in `message`?

Reply:
Is not used right now, but this should be used to send debugging messages. I'm not sure how I can properly debug the firmware in combination with the .py script.

Reply:
I'd oppose implementing a field purely for debugging purposes – the semantics of the messages should be elicited from the protobuf to draw clear separation of concerns, IMO.

Reply:
Keep it as is.